Disable -Werror in cvitek modules for GCC 14

GCC 14 introduces new warnings that are treated as errors.
Add -Wno-error globally for these vendor kernel modules.

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,7 @@
 SHELL=/bin/bash
+# Disable -Werror for GCC 14 compatibility
+export KCFLAGS += -Wno-error
+
 -include $(BUILD_PATH)/.config
 #
 export CVIARCH_L := $(shell echo $(CVIARCH) | tr A-Z a-z)
